airbyte-integrations/connectors/source-hubspot/source_hubspot/components.py

```diff
@@ -49,3 +54,69 @@ def transform(
                     else:
                         if record_or_schema.get(transformed_field) is None:
                             record_or_schema[transformed_field] = value
+
+
+@dataclass
+class ListFieldToSingleRecordByPkExtractor(DpathExtractor):
+    pk_field_path: List[Union[InterpolatedString, str]] = None
+    related_field_path: List[Union[InterpolatedString, str]] = None
+
+    def __post_init__(self, parameters: Mapping[str, Any]) -> None:
+        super().__post_init__(parameters)
+        self._pk_field_path = [InterpolatedString.create(path, parameters=parameters) for path in self.pk_field_path]
+        for path_index in range(len(self.pk_field_path)):
+            if isinstance(self.pk_field_path[path_index], str):
+                self._pk_field_path[path_index] = InterpolatedString.create(self.pk_field_path[path_index], parameters=parameters)
+
+        self._related_field_path = [InterpolatedString.create(path, parameters=parameters) for path in self.related_field_path]
+        for path_index in range(len(self.related_field_path)):
+            if isinstance(self.related_field_path[path_index], str):
+                self._related_field_path[path_index] = InterpolatedString.create(self.related_field_path[path_index], parameters=parameters)
+
+    def extract_records(
+        self,
+        response: requests.Response,
+    ) -> Iterable[Mapping[str, Any]]:
+        pk_field_path = [path.eval(self.config) for path in self._pk_field_path]
+        related_field_path = [path.eval(self.config) for path in self._related_field_path]
+
+        for record in super().extract_records(response):
+            pk_field_value = dpath.get(record, pk_field_path, default="")
+            related_field_values = dpath.get(record, related_field_path, default=[])
+
+            for related_field_value in related_field_values:
+                updated_record = record.copy()
+                updated_record = dpath.new(updated_record, pk_field_path, pk_field_value)
+                updated_record = dpath.new(updated_record, related_field_path, related_field_value)
+                updated_record = dpath.new(updated_record, related_field_path + pk_field_path, pk_field_value)
+                yield updated_record
+
+
+@dataclass
+class UnnestFields(RecordTransformation):
```

Comment:
this can be removed after this changes are merged https://github.com/airbytehq/airbyte-python-cdk/pull/472